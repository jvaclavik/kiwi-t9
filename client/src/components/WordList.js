// @flow
import React from "react"
import {
  ActivityIndicator,
  ScrollView,
  Text,
  TouchableHighlight,
  View,
} from "react-native"

// external libs
import shortid from "shortid"

// utils
import renderStyle from "../utils/renderStyle"

// theme
import { Colors, Metrics, Fonts } from "../themes"

const styles = {
  scrollView: {
    backgroundColor: Colors.light,
  },
  container: loading => ({
    justifyContent: loading ? "center" : "flex-start",
    flexGrow: 1,
  }),
  wordContainer: {
    borderRightColor: Colors.gray,
    borderRightWidth: 1,
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    paddingHorizontal: Metrics.spacings.large,
  },
  word: {
    alignItems: "center",
    color: Colors.text,
    fontSize: Fonts.sizes.subtitle,
  },
}

type Props = {|
  +words: Array<string>,
  +onWordPress: string => void,
  +loading: boolean,
|}

export default class WordList extends React.PureComponent<Props> {
  render() {
    const { loading, onWordPress, words } = this.props
    return (
      <ScrollView
        contentContainerStyle={renderStyle(styles.container, loading)}
        horizontal
        style={renderStyle(styles.scrollView)}
      >
        {loading ? (
          <ActivityIndicator size="small" color={Colors.text} />
        ) : (
          words.map(word => (
            <TouchableHighlight
              underlayColor={Colors.gray}
              onPress={() => onWordPress(word)}
              key={shortid.generate()}
            >
              <View style={renderStyle(styles.wordContainer)}>
                <Text style={renderStyle(styles.word)}>{word}</Text>
              </View>
            </TouchableHighlight>
          ))
        )}
      </ScrollView>
    )
  }
}
