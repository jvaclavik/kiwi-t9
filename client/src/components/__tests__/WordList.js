// @flow
import React from "react"
import renderer from "react-test-renderer"

// components
import WordList from "../WordList"

test("renders correctly with required props only", () => {
  const tree = renderer
    .create(
      <WordList
        words={["hello", "hola", "ahojky"]}
        loading={false}
        onWordPress={() => null}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders correctly with optional props", () => {
  const tree = renderer
    .create(
      <WordList
        words={["hello", "hola", "ahojky"]}
        loading
        style={{ padding: 10 }}
        onWordPress={() => null}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
