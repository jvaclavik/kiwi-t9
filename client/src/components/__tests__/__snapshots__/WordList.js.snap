// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with optional props 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
/>
`;

exports[`renders correctly with required props only 1`] = `
<View
  style={Object {}}
/>
`;
